WordPress/Classes/ViewRelated/Post/PostListViewController.swift

```diff
@@ -0,0 +1,742 @@
+            return self.dynamicType.postCardRestoreCellRowHeight
+        }
+        
+        var cell : PostCardTableViewCell
+        
+        if post.pathForDisplayImage?.characters.count > 0 {
+            cell = textCellForLayout
+        } else {
+            cell = imageCellForLayout
+        }
+        
+        configureCell(cell, atIndexPath: indexPath)
+        let size = cell.sizeThatFits(CGSizeMake(width, CGFloat.max))
+        let height = ceil(size.height)
+        
+        return height
+    }
+    
+    func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
+        tableView.deselectRowAtIndexPath(indexPath, animated: true)
+        
+        guard let post = tableViewHandler?.resultsController.objectAtIndexPath(indexPath) as? AbstractPost else {
+            let message = "Expected a post object."
+            assertionFailure(message)
+            DDLogSwift.logError("\(#file): \(#function) [\(#line)] - \(message)")
+            return
+        }
+
+        if post.remoteStatus == AbstractPostRemoteStatusPushing {
+            // Don't allow editing while pushing changes
+            return
+        }
+        
+        if post.status == PostStatusTrash {
+            // No editing posts that are trashed.
+            return
+        }
+        
+        previewEditPost(post)
+    }
+    
+    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        guard let post = tableViewHandler?.resultsController.objectAtIndexPath(indexPath) as? Post else {
+            preconditionFailure("Expected a post object.")
+        }
+        
+        let identifier = cellIdentifierForPost(post)
+        let cell = tableView.dequeueReusableCellWithIdentifier(identifier, forIndexPath: indexPath)
+        
+        configureCell(cell, atIndexPath: indexPath)
+        
+        return cell
+    }
+    
+    func configureCell(cell: UITableViewCell, atIndexPath indexPath: NSIndexPath) {
+        cell.accessoryType = .None
+        cell.selectionStyle = .None
+        
+        if let postCell = cell as? PostCardCell {
+            postCell.delegate = self
+            
+            if let post = tableViewHandler?.resultsController.objectAtIndexPath(indexPath) as? WPPostContentViewProvider {
+                let layoutOnly = (cell == imageCellForLayout) || (cell == textCellForLayout)
+                
+                postCell.configureCell?(post, layoutOnly: layoutOnly)
+            }
+        }
+    }
+    
+    private func cellIdentifierForPost(post: Post) -> String {
+        var identifier : String
+        
+        if recentlyTrashedPostObjectIDs?.containsObject(post.objectID) == true && currentPostListFilter()?.filterType != .Trashed {
+            identifier = self.dynamicType.postCardRestoreCellIdentifier
+        } else if post.pathForDisplayImage?.characters.count > 0 {
```

Comment:
This logic is inverted from the original, if you have a path for a image you should display an image cell, if it's just text a text cell.